src/archive/extract.rs

```diff
@@ -1,6 +1,7 @@
 use std::error::Error as StdError;
 use std::path::Path;
 
+#[allow(clippy::enum_variant_names)]
```

Comment:
Warnings:

```
warning: variant name ends with the enum's name
 --> src/archive/extract.rs:6:5
  |
6 |     IoError(std::io::Error),
  |     ^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(clippy::enum_variant_names)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names

warning: all variants have the same postfix: `Error`
 --> src/archive/extract.rs:5:1
  |
5 | / pub enum Error {
6 | |     IoError(std::io::Error),
7 | |     ZipError(zip::result::ZipError),
8 | |     HttpError(reqwest::Error),
9 | | }
  | |_^
  |
  = help: remove the postfixes and use full paths to the variants instead of glob imports
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names
```

I'm leaving this to the author in this PR because I'm not so good at this repo currently.
What do you think? @TaKO8Ki